Clean up `pins` module

hannobraun opened the pull request.

david-sawatzke commented on src/pins/state.rs:
```diff
@@ -0,0 +1,67 @@
+//! Contains types that indicate pin states
+//!
+//! Please refer to [`Pin`] for documentation about how these types are used.
```
`Pin` isn't linked (compared to all the other links to `Pin` in this file)

david-sawatzke commented on src/pins/state.rs:
```diff
@@ -0,0 +1,67 @@
+/// documentation, to show which states a pin can be in. Other than that,
+/// this trait should not be relevant to users of this crate.
+///
+/// [`Pin`]: ../struct.Pin.html
```
I think this link isn't valid anymore

david-sawatzke replied:
(also applies to all the other links to `Pin`)

hannobraun replied:
It is actually, due to the public re-export in `pins`. That's what I think anyway, but the link does work in the generated output.

hannobraun commented:
I fixed all broken links in the `pins` documentation and updated some of it. Merging, since the previous commits were already approved, and the new commits only change documentation, which is largely outdated and in need of an update anyways.
